cluster/saltbase/salt/kube-master-addons/kube-master-addons.sh

```diff
@@ -15,52 +15,73 @@
-
-    timeout 30 docker load -i /srv/salt/kube-bins/kube-apiserver.tar 1>/dev/null 2>&1
-    rc=$?
-    if [[ $rc == 0 ]]; then
-      let loadedImageFlags="$loadedImageFlags|1"
-    elif [[ $rc == 124 ]]; then
-      restart_docker=true
-    fi
-
-    timeout 30 docker load -i /srv/salt/kube-bins/kube-scheduler.tar 1>/dev/null 2>&1
-    rc=$?
-    if [[ $rc == 0 ]]; then
-      let loadedImageFlags="$loadedImageFlags|2"
-    elif [[ $rc == 124 ]]; then
-      restart_docker=true
+function load-docker-images() {
+  let loadedImageFlags=0
+  
+  while true; do
+    restart_docker=false
+  
+    if which docker 1>/dev/null 2>&1; then
+  
+      timeout 30 docker load -i /srv/salt/kube-bins/kube-apiserver.tar 1>/dev/null 2>&1
+      rc=$?
+      if [[ $rc == 0 ]]; then
+        let loadedImageFlags="$loadedImageFlags|1"
+      elif [[ $rc == 124 ]]; then
+        restart_docker=true
+      fi
+  
+      timeout 30 docker load -i /srv/salt/kube-bins/kube-scheduler.tar 1>/dev/null 2>&1
+      rc=$?
+      if [[ $rc == 0 ]]; then
+        let loadedImageFlags="$loadedImageFlags|2"
+      elif [[ $rc == 124 ]]; then
+        restart_docker=true
+      fi
+  
+      timeout 30 docker load -i /srv/salt/kube-bins/kube-controller-manager.tar 1>/dev/null 2>&1
+      rc=$?
+      if [[ $rc == 0 ]]; then
+        let loadedImageFlags="$loadedImageFlags|4"
+      elif [[ $rc == 124 ]]; then
+        restart_docker=true
+      fi
     fi
-
-    timeout 30 docker load -i /srv/salt/kube-bins/kube-controller-manager.tar 1>/dev/null 2>&1
-    rc=$?
-    if [[ $rc == 0 ]]; then
-      let loadedImageFlags="$loadedImageFlags|4"
-    elif [[ $rc == 124 ]]; then
-      restart_docker=true
+  
+    # required docker images got installed. exit while loop.
+    if [[ $loadedImageFlags == 7 ]]; then break; fi
+  
+    # Sometimes docker load hang, restart docker daemon resolve the issue
+    if [[ $restart_docker ]]; then
+      if ! service docker restart; then # Try systemctl if there's no service command.
+        systemctl restart docker      
+      fi
     fi
-  fi
+  
+    # sleep for 15 seconds before attempting to load docker images again
+    sleep 15
+  
+  done
+}
 
-  # required docker images got installed. exit while loop.
-  if [[ $loadedImageFlags == 7 ]]; then break; fi
+function convert-rkt-image() {
+  (cd /tmp; ${DOCKER2ACI_BIN} $1)
+}
 
-  # Sometimes docker load hang, restart docker daemon resolve the issue
-  if [[ $restart_docker ]]; then
-    if ! service docker restart; then # Try systemctl if there's no service command.
-      systemctl restart docker      
-    fi
-  fi
+function load-rkt-images() {
+  convert-rkt-image /srv/salt/kube-bins/kube-apiserver.tar
+  convert-rkt-image /srv/salt/kube-bins/kube-scheduler.tar
+  convert-rkt-image /srv/salt/kube-bins/kube-controller-manager.tar
 
-  # sleep for 15 seconds before attempting to load docker images again
-  sleep 15
+  ${RKT_BIN} fetch /tmp/*.aci --insecure-options=image
```

Comment:
It would be nice if there was a better way to do this. E.g. if `fetch` could read a local docker tarball (per https://github.com/coreos/rkt/issues/2392).

Maybe a comment explaining why this hackery is needed (combination of docker2aci not having a well-known output file name + rkt support in that issue).